strapi/strapi
# 2FA (two factor authentication) via email for admin panel user accounts

### What does it do?

When enabled, users will receive a 2FA code to their registered email upon logging in. They will then be redirected to a new page to enter the 2FA code received via email.

### Why is it needed?
There has been a growing interest in implementing 2FA for admin accounts over the years.
[feedback forum](https://feedback.strapi.io/security/p/support-2fa-via-users-permissions-plugin)
[Issue #786](https://github.com/strapi/strapi/issues/786)
There are also quite a bit of questions regarding it on the official discord channel

### How to test it?

Ensure that your email system is properly configured to receive password reset emails, etc.
Enable the settings under the admin panel's advanced settings section labelled under: "Enable multi-factor authentication"
Logout and log back in. Wait for 2FA code to be emailed to your inbox registered with Strapi account. Enter the 2FA code and get authenticated.

### Related issue(s)/PR(s)

[Issue #786](https://github.com/strapi/strapi/issues/786)


### Comments

Hi Strapi team :wave: 
My name is Leo and this is my first PR to the project. Recently, I had the need for multi factor authentication to the admin user accounts for enhanced security. I was disappointed to find that this was not already supported in the latest version. Alas, I decided to add the implementation myself with [patch-package](https://www.npmjs.com/package/patch-package) and [postinstall](https://www.npmjs.com/package/postinstall). After seeing it work quite nicely for a simple use, I wanted to share my solution with the community. I would like to get an initial feedback from the maintainers to see if my approach is okay. If approved, I would like to add some more testing for complete the PR requirements.
